src/permutation.rs

```diff
@@ -96,6 +93,110 @@ impl Permutation {
         debug_assert!(result.valid());
         return result;
     }
+    /// Computes the permutation that would sort a given slice.
+    ///
+    /// This calculates the permutation that if it were applied to the slice,
+    /// would put the elements in sorted order.
+    ///
+    /// # Panics
+    ///
+    /// If self.len() != vec.len()
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use permutation::Permutation;
+    /// let mut vec = vec!['z','w','h','a','s','j'];
+    /// let mut permutation = permutation::Permutation::one(vec.len());
+    /// permutation.sort(&vec[..]);
+    /// let permuted = permutation.apply_slice(&vec[..]);
+    /// vec.sort();
+    /// assert_eq!(vec, permuted);
+    /// ```
+    ///
+    /// You can also use it to sort multiple arrays based on the ordering of one.
+    ///
+    /// ```
+    /// let names = vec!["Bob", "Steve", "Jane"];
+    /// let salary = vec![10, 5, 15];
+    /// let mut permutation = permutation::Permutation::one(salary.len());
+    /// permutation.sort(&salary[..]);
+    /// let ordered_names = permutation.apply_slice(&names[..]);
+    /// let ordered_salaries = permutation.apply_slice(&salary[..]);
+    /// assert_eq!(ordered_names, vec!["Steve", "Bob", "Jane"]);
+    /// assert_eq!(ordered_salaries, vec![5, 10, 15]);
+    /// ```
+    pub fn sort<T, D>(&mut self, vec: D)
+    where
+        T: Ord,
+        D: Deref<Target = [T]>,
+    {
+        assert_eq!(self.len(), vec.len());
+        //We use the reverse permutation form, because its more efficient for applying to indices.
+        self.indices.sort_by_key(|&i| &vec[i]);
+    }
+
+    /// Computes the permutation that would sort a given slice by a comparator.
+    ///
+    /// This is the same as `permutation::sort()` except that it allows you to specify
+    /// the comparator to use when sorting similar to `std::slice.sort_by()`
+    ///
+    /// # Panics
+    ///
+    /// If self.len() != vec.len()
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use permutation::Permutation;
+    /// let mut vec = vec!['z','w','h','a','s','j'];
+    /// let mut permutation = permutation::Permutation::one(vec.len());
+    /// permutation.sort_by(&vec[..], |a, b| b.cmp(a));
```

Comment:
Maybe add a comment that the permutation is reused without another allocation?  Also in the example below.

Reply:
Will do.